chapter10/index.adoc

```diff
@@ -0,0 +1,484 @@
+
+Built on top of Jakarta EE and tightly integrated with other MicroProfile APIs, MicroProfile JWT allows developers to build secure, interoperable, and portable microservices. It supports role-based access control (RBAC), simplifies identity management in stateless services, and avoids vendor lock-in by adhering to open specifications.
+
+=== Topics to be covered:
+- Introduction to JWT Authentication
+- Structure of JWT
+- Benefits of JWT in Microservice? 
+- Setting up MicroProfile JWT
+- Configuring MicroProfile JWT
+- Use cases for JSON Web Tokens
+- Request Flow in MicroProfile JWT
+- Role-Based Access Control (RBAC)
+- Setting Token Expiry Times for Security
+- Error Handling
+- Best Practices for JWT Authentication
+- Security Best Practices for Microservices
+- Conclusion
+
+=== Introduction to JWT Authentication
+
+In this section, we will explore what JSON Web Tokens are, how they work, and why they are foundational to implementing stateless authentication and authorization in microservices-based systems.
+
+==== What is a JSON Web Token (JWT)?
+
+https://jwt.io/[JSON Web Token (JWT)], defined by https://datatracker.ietf.org/doc/html/rfc7519[RFC 7519] is an open standard for securely transmitting information (claims) between parties as a JSON object. JWTs are digitally signed, ensuring their integrity and authenticity.
+
+=== Structure of a JWT
+
+A JWT consists of three Base64 encoded parts, separated by dots (+.+): 
+
+[source]
+----
+<Header>.<Payload>.<Signature>
+----
+
+- *Header* -  It contains metadata about the token, such as token type (type: “JWT”) and signing algorithm ( alg: “RS256” for RSA-SHA256). 
+
+[source, json]
+----
+{
+  "alg": "RS256",
+  "typ": "JWT"
+}
+----
+- *Payload* - It contains claims which are key-value pairs representing data about the user such as roles and expiration.
+Example of claims in a JWT payload:
+
+[source, json]
+----
+{
+  "iss": "https://io.microprofile.com/issuer",
+  "sub": "user1",
+  "exp": 1735689600,
+  "iat": 1735686000,
+  "aud": "my-audience",
+  "groups": ["user", "admin"]
+}
+----
+
+- *Signature* - A digital signature that is used to verify the token’s integrity by combining the encoded header, payload, and a secret/private key.
+
+Example JWT Token:
+[source]
+----
+eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJ1c2VyMTIzIiwiaXNzIjoiaHR0cHM6Ly9hdXRoLmV4YW1wbGUuY29tIiwiZ3JvdXBzIjpbInVzZXIiLCJhZG1pbiJdfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c
+----
+
+==== Types of Claims in MicroProfile JWT
+
+Claims in JWTs can be categorized into two types:
+
+===== Standard Claims
+
+These are predefined claims with specific meanings, as defined by the JWT specification. MicroProfile JWT recognizes and validates these claims during token validation. Some commonly used standard claims include:
```

Comment:
```suggestion
These are predefined claims with specific meanings, as defined by the JWT specification. Some commonly used standard claims include:
```

MP JWT does not currently deal with the `jti` and `nbf` claims.

Perhaps, `MicroProfile JWT recognizes and validates these claims during token validation.` can remain but then please remove these 2 claims from the table below